src/outputs/output_utils.hpp

```diff
@@ -0,0 +1,212 @@
+//========================================================================================
+// Parthenon performance portable AMR framework
+// Copyright(C) 2020-2023 The Parthenon collaboration
+// Licensed under the 3-clause BSD License, see LICENSE file for details
+//========================================================================================
+// (C) (or copyright) 2020-2023. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#ifndef OUTPUTS_OUTPUT_UTILS_HPP_
+#define OUTPUTS_OUTPUT_UTILS_HPP_
+
+// C++
+#include <array>
+#include <map>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <tuple>
+#include <vector>
+
+// Parthenon
+#include "interface/metadata.hpp"
+#include "interface/variable.hpp"
+#include "mesh/mesh.hpp"
+#include "mesh/meshblock.hpp"
+#include "utils/error_checking.hpp"
+
+namespace parthenon {
+namespace OutputUtils {
```

Comment:
I agree that contiguous particle data is best. I'm not immediately succeeding in parsing the output logic before (I trust that it works because outputs work). But my only question is -- is it possible to easily get out just the particles from a particular block, maybe just at the `phdf` level?

Reply:
Yes, that's possible. I could add some `phdf` machinery to do this if you want. It's contained in the `offset` and `count` array. The particles in block `b` are given by the `swarm/SwarmVar/var[offsets[b]:offsets[b]+counts[b]+1]`

Reply:
Ok you can now do the following
```python
slc = swarm.Block(b)
print(Swarm.x[slc]) # prints only the x components of particles on block b
```

Reply:
The Ascent output needs an update in order to build after the change from `VarInfo` -> `OutputUtils::VarInfo`.

Reply:
I think the only change needed is on line `ascent.cpp:170`.